django/core/management/commands/dumpdata.py

```diff
@@ -144,7 +144,16 @@ def get_objects(count_only=False):
             Collate the objects to be serialized. If count_only is True, just
             count the number of objects to be serialized.
             """
-            models = serializers.sort_dependencies(app_list.items())
+            if use_natural_foreign_keys:
+                models = serializers.sort_dependencies(app_list.items(), allow_loops=True)
+
+            else:
+                models = []
+                for (app_config, app_models) in app_list.items():
+                    if app_models == None:
+                        models.extend(app_config.get_models())
+                    else:
+                        models.extend(app_models)
```

Comment:
Any reason why `serializers.sort_dependencies(app_list.items(), allows_cycles=False)` wouldn't work here?

If that's the case then this whole chunk could be replaced by

```python
serializers.sort_dependencies(
    app_list.items(), allow_cycles=use_natural_foreign_keys
)

Reply:
That would also work, but do sorting work that is not useful at all. I guess it just comes down to efficiency then.